### ISSUE TITLE
General performance, node-spdy vs node-http2

### ISSUE THREAD

**Issue by KenanSulayman** [molnarg/node-http2#98]
Hey there,

I just thought about rolling out http2, so I tried it on my own servers.

I am experiencing extreme performance loss compared to node-spdy, which I used before: is this a known issue?

I setup a http2 reference server, so that you can see it yourself:
 - **node-spdy**: [https://sly.mn:443](https://sly.mn:443)
 - **node-http2**: [https://sly.mn:10080](https://sly.mn:10080)

**node-spdy** loading time:

![spdy](https://cloud.githubusercontent.com/assets/280212/6263499/f3179e5e-b818-11e4-82f8-582f851a75d3.png)

**node-http2** loading time:

![http2](https://cloud.githubusercontent.com/assets/280212/6263527/4d2064bc-b819-11e4-897c-01f7c5e1f3e2.png)

**Comment by molnarg** [comment]
Hi, no this not a known issue. Node-http2 is not yet optimized for performance, but this is certainly more than the lack of optimizations; something wrong is going on here.

Is the other part of your server code (reading/caching files, etc.) identical to your SPDY server code?

**Comment by KenanSulayman** [comment]
Yes, the code is open-sourced as well. To make this easier for you:

 - [main server](https://github.com/KenanSulayman/LEY/blob/master/app.js#L372-L502) (assets excl. video)
- [subroutine for streaming the background video](https://github.com/KenanSulayman/LEY/blob/master/controller/stream.js#L21-L87)

Difference between the servers I setup:
 - both *htto2*  & *node-10-http2*: https://gist.github.com/KenanSulayman/4b02902ef41933e5c32c for changing the ports
 - drop-in replaced spdy with http2, just replaced, that is. Same code.

Cheers

**Comment by vigneshshanmugam** [comment]
https://sly.mn:10080 is running in https mode not in http2.
